assignments/arrays.js

```diff
@@ -62,33 +62,79 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
+function carCheck(obj,id){
+    for ( let i = 0 ; i < obj.length ;i++){
+        if(obj[i].id === id) {
+            console.log(`Car ${id} is a ${obj[i].car_year} ${obj[i].car_make} ${obj[i].car_model}` );
+            }
+        }
+    
+    }
+    
+    carCheck(inventory,33);
+    console.log("===========================================================")
+    
+    // ==== Challenge 2 ====
+    // The dealer needs the information on the last car in their inventory.  
+    //What is the make and model of the last car in the inventory?  Log the make and model into the console.
+    
+    carCheck(inventory,inventory.length);
+    // ==== Challenge 3 ====
+    // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into
+    // alphabetical order and log the results in the console
+    
+    let carModels = [];
+    for (let i = 0 ; i < inventory.length ;i++){
+    carModels.push(inventory[i].car_make);
+    }
+    carModels = carModels.sort();
+    console.log("===========================================================")
+    console.log("Car models are ");
+    for ( i in carModels){
+        console.log(carModels[i]);
+    }
+    // ==== Challenge 4 ====
+    // The accounting team needs all the years from every car on the lot. Create a new array from the 
+    //dealer data containing only the car years and log the result in the console.
+    console.log("===========================================================")
+    let carYears = [];
+    for (let i = 0 ; i < inventory.length ;i++){
+    carYears.push(inventory[i].car_year);
+    }
+    console.log("The list of years is \n" + carYears+"\n");
+    console.log("===========================================================")
+    // ==== Challenge 5 ====
+    // The car lot manager needs to find out how many cars are older than the year 2000. 
+    //Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array 
+    //oldCars and logging it's length.
+    console.log("===========================================================")
+    let oldCars =[];
+    for ( let i = 0 ; i < carYears.length ; i++){
+            if(carYears[i] <2000){
+                oldCars.push(carYears[i]);
+            }
+    }
+    
+    console.log("the list of old cars is \n" + oldCars); 
+    console.log("===========================================================")
+    // ==== Challenge 6 ====
+    // A buyer is interested in seeing only BMW and Audi cars within the inventory.  
+    //Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, 
+    //use JSON.stringify() to show the results of the array in the console.
+    let BMWAndAudi=[];
+    for ( let i = 0 ; i<inventory.length ; i++){
+            let make = inventory[i].car_make.toLowerCase();
+            if(make ==="audi" || make ==="bmw"){
+                BMWAndAudi.push(make);
```

Comment:
this function works ALMOST how you want it to, but you are only pushing the car_make to the new array, and that doesn't seem very useful. it would be better to push ALL the info about that car.

```BMWAndAudi.push(inventory[i]);```